core/libp2p/multi/multihash.cpp

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/multi/multihash.hpp"
+#include <boost/algorithm/hex.hpp>
+#include <boost/format.hpp>
+#include "common/hexutil.hpp"
+
+using kagome::expected::Error;
+using kagome::expected::Result;
+using kagome::expected::Value;
+
+using kagome::common::Buffer;
+using kagome::common::unhex;
+
+namespace libp2p::multi {
+
+  Multihash::Multihash(const HashType &type, Hash hash)
+      : data_{Buffer({type, (uint8_t)hash.size()}).put(hash.toVector())},
```

Comment:
better add range constructor to Buffer and write 
```
data_{hash.begin(), hash.end()}
```

Reply:
The problem is I need to put in data_ a buffer consisting of a type, the length and the hash, so I don't see how I can use the copy constructor here.

Reply:
Ah I see